pycqed/measurement/waveform_control/pulse.py

```diff
@@ -173,6 +208,15 @@ def hashables(self, tstart, channel):
         raise NotImplementedError('hashables() not implemented for {}'
                                   .format(str(type(self))[1:-1]))
 
+    def common_hashables(self, tstart, channel):
```

Comment:
I don't know what this function does so I can't really review this part. But I think this has been used already on the measurement setups so it should be fine.

Reply:
The parameters of the truncation feature in pulse must be part of the hashables of every pulse. This is required to ensure that reuse_waveforms and the sequence caching in pulse work correctly when the truncation feature is used.